### PR TITLE
i2p: cleanup and build jbigi from source

### PR DESCRIPTION
**Pull request by linsui**
## Description of changes

We have service.i2p and the i2p daemon wrapper is never used. So I removed all related files. Also build jbigi from source.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/i2p/default.nix [hunk 1]
```diff
@@ -1,60 +1,98 @@
...
+    urls = map (mirror: "${mirror}${finalAttrs.version}/i2psource_${finalAttrs.version}.tar.bz2") [
+      "https://github.com/i2p/i2p.i2p/releases/download/i2p-"
+      "https://download.i2p2.de/releases/"
+      "https://files.i2p-projekt.de/"
+      "https://download.i2p2.no/releases/"
     ];
-    sha256 = "sha256-MO+K/K0P/6/ZTTCsMH+GtaazGOLB9EoCMAWEGh/NB3w=";
+    hash = "sha256-JfPjjNVtXg5LCpg+k2973WxFmsChOyUMPGb8v4J8fRw=";
+    # Remove unused binaries
+    postFetch = "rm -r $out/installer/lib";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/i2p/default.nix:27]: This makes it harder to reproduce the download entry. Is there another reason than cleanup to do this? If not doing this in postUnpack would be nicer.

File: pkgs/tools/networking/i2p/default.nix [hunk 2]
```diff
@@ -1,60 +1,98 @@
...
 
   buildPhase = ''
+    # When this variable exists we can build the .so files only.
+    export DEBIANVERSION=1
+    pushd core/c/jcpuid
+    ./build.sh
+    popd
+    pushd core/c/jbigi
+    ./build_jbigi.sh dynamic
+    ls
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/i2p/default.nix:58]:
```suggestion
```

File: pkgs/tools/networking/i2p/default.nix [hunk 3]
```diff
@@ -1,60 +1,98 @@
...
-      -e "s#%JAVA%#${jre}/bin/java#"
-    mv $out/runplain.sh $out/bin/i2prouter-plain
+    mkdir -p $out/{bin,share,geoip}
+    mv pkg-temp/* $out
+    mv core/c/jbigi/*.so $out/lib
     mv $out/man $out/share/
-    chmod +x $out/bin/* $out/i2psvc
     rm $out/{osid,postinstall.sh,INSTALL-headless.txt}
+
+    for jar in $(ls $out/lib/*.jar); do
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/i2p/default.nix:71]:
```suggestion
    for jar in $out/lib/*.jar; do
```

File: pkgs/tools/networking/i2p/default.nix (hunk 1 above)
Comment (linsui) [reply pkgs/tools/networking/i2p/default.nix:27]: I want to make the src tarball smaller. Since it's not preferred I removed this.

**Pull request merged**